agent/engine/docker_task_engine.go

```diff
@@ -1238,12 +1240,139 @@ func (engine *DockerTaskEngine) GetDaemonManagers() map[string]dm.DaemonManager
 func (engine *DockerTaskEngine) pullContainerManifest(
 	task *apitask.Task, container *apicontainer.Container,
 ) dockerapi.DockerContainerMetadata {
-	// Currently a no-op
-	logger.Debug("Manifest pull is currently a no-op", logger.Fields{
-		field.TaskID:    task.GetID(),
-		field.Container: container.Name,
-		field.Image:     container.Image,
+	switch container.Type {
+	case apicontainer.ContainerCNIPause,
+		apicontainer.ContainerNamespacePause,
+		apicontainer.ContainerServiceConnectRelay,
+		apicontainer.ContainerManagedDaemon:
+		// pause images and AppNet relay image are managed at startup
+		return dockerapi.DockerContainerMetadata{}
+	}
+	// AppNet Agent container image is also managed at start up
+	// (it uses the same image as AppNet Relay container)
+	if task.IsServiceConnectEnabled() && container == task.GetServiceConnectContainer() {
+		return dockerapi.DockerContainerMetadata{}
+	}
+
+	var imageManifestDigest digest.Digest
+	digestFromPayload := referenceutil.GetDigestFromImageRef(container.Image)
+	if digestFromPayload != "" {
+		// Digest available in task payload
+		imageManifestDigest = digestFromPayload
+		logger.Info("Found image manifest digest in task payload", logger.Fields{
+			field.TaskARN:       task.Arn,
+			field.ContainerName: container.Name,
+			field.Image:         container.Image,
```

Comment:
nit: I guess it's a part of the image field, but could we log the digest as a field here in case we ever make a mistake parsing the digest and need to verify if we got the correct one from the image?

Reply:
Good point. I will make this change.

Reply:
Updated.